src/utils/metadata.ts

```diff
@@ -10,6 +10,66 @@ export function transformTags(value: string): string[] {
   return transformedTags
 }
 
+export function mapTimeoutStringToSeconds(timeout: string): number {
+  switch (timeout) {
+    case 'Forever':
+      return 0
+    case '1 day':
+      return 86400
+    case '1 week':
+      return 604800
+    case '1 month':
+      return 2630000
+    case '1 year':
+      return 31556952
+    default:
+      return 0
+  }
+}
+
+function numberEnding(number: number): string {
+  return number > 1 ? 's' : ''
+}
+
+export function millisecondsToStr(milliseconds: number): string {
+  if (milliseconds === 0) return 'Forever'
+
+  let temp = Math.floor(milliseconds / 1000)
+  const years = Math.floor(temp / 31536000)
+  const weeks = Math.floor((temp %= 31536000) / 604800)
+  const days = Math.floor((temp %= 604800) / 86400)
+  const hours = Math.floor((temp %= 86400) / 3600)
+  const minutes = Math.floor((temp %= 3600) / 60)
+  const seconds = temp % 60
+
+  return years
+    ? `${years} year${numberEnding(years)}`
+    : weeks
+    ? `${weeks} week${numberEnding(weeks)}`
+    : days
+    ? `${days} day${numberEnding(days)}`
+    : hours
+    ? `${hours} hour${numberEnding(hours)}`
+    : minutes
+    ? `${minutes} minute${numberEnding(minutes)}`
+    : seconds
+    ? `${seconds} second${numberEnding(seconds)}`
+    : 'less than a second'
+}
+
+export function checkIfTimeoutInPredefinedValues(timeout: string): boolean {
+  const predefinedValues = ['Forever', '1 day', '1 week', '1 month', '1 year']
```

Comment:
I feel we are having this list in too many places by now and we should be able to keep it down to 2 content files, for publish & edit form (ideally we would only have 1 content file for this but this was tricky to do). So the list in the content file should be the source of truth for this list instead of repeating it within the code, which opens up possible maintainability mistakes.

So in this case we could either import the content json directly in here (with downside that whole file is then included in js bundle), or accept `predefinedValues` as a method parameter and pass them from the component where we get the content with Gatsby's GraphQL

Reply:
tinkered around with this and realized we already have the actual content, with the input field definition and the timeout options, passed down to component already so I quickly refactored it to make use of that. So I was able to remove this hardcoded list in https://github.com/oceanprotocol/market/pull/324/commits/1391d2663dd0b5185dd49393eb642d2fc93c86eb

Now we are down to 3 places (where 2 are purely content files) we would have to update, ideally we only have to update the 2 content files and the app does the rest. But not sure we can incorporate that timeout options list into that `mapTimeoutStringToSeconds()` method https://github.com/oceanprotocol/market/pull/324/files#diff-2ec51fbdbf76b95773fe87b47bbac25114e58f14be9da6f2ef5772fb6299b9c8R13

If anybody has an idea to accomplish this please go ahead, otherwise I think it's good enough and we keep that in mind if we update the list